src/components/gauge/gauge.tsx

```diff
@@ -0,0 +1,381 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Listen,
+  Prop,
+  State
+} from "@stencil/core";
+
+import { IComponent } from "../common/interfaces";
+
+@Component({
+  shadow: false,
+  styleUrl: "gauge.scss",
+  tag: "gx-gauge"
+})
+export class Gauge implements IComponent {
+  @Element() element: HTMLElement;
+
+  @Event() gxGaugeDidLoad: EventEmitter;
+
+  /**
+   * Property of type Style.
+   * Define if shadow will display or not. Default is disabled.
+   */
+  @Prop() styleShadow = false;
+
+  /**
+   * This property allows you to select the gauge type. _(Circle or Line)_.
+   * Default is linear type.
+   */
+  @Prop() gaugeType: "line" | "circle" = "line";
+
+  /**
+   *  Allows display current value. Default is disabled.
+   *
+   */
+  @Prop() showValue = false;
+
+  /**
+   * The Minimum Value of the gauge
+   *
+   */
+  @Prop() minValue: number;
+
+  /**
+   * The Currecnt value indicanding in the gauge
+   *
+   */
+  @Prop() currentValue: number = this.minValue;
+
+  /**
+   * This allows specifying the width of the circumference _(When gauge is Circle type)_ and the width of the bar _(When gauge is Line type)_ in % relative the component size.
+   *
+   */
+  @Prop() thickness = 10;
+
+  @State() maxValue: number;
+
+  @State() totValues = this.minValue;
+
+  @State() children = [];
+
+  @State() minorSize: number;
+
+  @Listen("gxGaugeRangeDidLoad")
+  onGaugeRangeDidLoad({ detail: childRange }) {
+    this.children = [...this.children, childRange];
+    this.totValues += childRange.amount;
+    childRange.element.addEventListener("gxGaugeRangeDidUnload", () => {
+      const index = this.children.findIndex(x => x === childRange);
+      this.children.splice(index, 1);
+      this.totValues -= childRange.amount;
+    });
+    childRange.element.addEventListener("gxGaugeRangeDidUpdate", () => {
+      const index = this.children.findIndex(x => x === childRange);
+      this.children.splice(index, 1, childRange);
+      this.totValues = 0;
+      for (const childInstance of this.children) {
+        this.totValues += childInstance.amount;
+      }
+    });
+  }
+
+  componentWillLoad() {
+    this.totValues = 0;
+  }
+
+  private calcThickness() {
+    return typeof this.thickness === "number" &&
+      (this.thickness > 0 && this.thickness <= 100)
+      ? this.thickness / 5
+      : 10;
+  }
+
+  private calcPercentage() {
+    return (
+      (this.currentValue - this.minValue) *
+      100 /
+      (this.maxValue - this.minValue)
+    );
+  }
+
+  private renderCircle(childRanges) {
+    const svgRanges = [];
+    let acumulation = 0;
+    //////////////////////////////////////////
+    function calcPositionRange(preValue, value) {
```

Comment:
Does this need a separate function? It's called only from one place to do a simple operation.

Reply:
Yeap, I think that it's necessary to. Because including the function logic in the place where it's called, the code will be hard to read. I think that is better in that way, separating the code for better understanding.